Packs/RSANetWitness_v11_1/Integrations/RSANetWitnessv115/RSANetWitnessv115.py

```diff
@@ -1151,6 +1173,238 @@ def get_now_time() -> str | None:
+
+    """
+    modified_incidents_ids = []
+    remote_args = GetModifiedRemoteDataArgs(args)
+    max_fetch_alerts = min(arg_to_number(params.get('max_alerts')) or DEFAULT_MAX_INCIDENT_ALERTS, DEFAULT_MAX_INCIDENT_ALERTS)
+    max_time_mirror_inc = min(arg_to_number(params.get("max_mirror_time")) or 3, 24)
+    last_update = remote_args.last_update
+    last_update_format = dateparser.parse(last_update, settings={'TIMEZONE': 'UTC'})  # converts to a UTC timestamp
+
+    demisto.debug(f'Running get-modified-remote-data command. Last update is: {last_update_format}')
+
+    # setting request
+    datetime_now = datetime.now()
+    since = datetime_now - timedelta(days=max_time_mirror_inc)
+    since_format = since.strftime(DATE_FORMAT)
+    until_format = datetime_now.strftime(DATE_FORMAT)
+    response, items = paging_command(MAX_NB_MIRROR_PULL, None, None, client.list_incidents_request,
+                                     until=until_format, since=since_format)
+
+    demisto.debug(f"Total Retrieved Incidents : {len(items)} in {response.get('totalPages')} pages")
+
+    # clean the integration context data of "old" incident
+    clean_old_inc_context(max_time_mirror_inc)
+    intCont = get_integration_context().get("IncidentsDataCount", {})
+    for inc in items:
+        if intCont.get(inc.get('id')):
+            demisto.debug(f"Last run incident {inc.get('id')} => "
+                          f"Alert count: {intCont.get(inc.get('id'), {}).get('alertCount')} "
+                          f"Event count: {intCont.get(inc.get('id'), {}).get('eventCount')}")
+            save_alert_count = intCont.get(inc.get('id'), {}).get('alertCount')
+            save_event_count = intCont.get(inc.get('id'), {}).get('eventCount')
+            if save_alert_count != inc.get('alertCount') or save_event_count != inc.get('eventCount'):
+                # compare the save nb of alert to see if we need to pull the alert or not
+                if save_alert_count <= max_fetch_alerts:
+                    modified_incidents_ids.append(inc.get("id"))
+                    continue  # if added no need to do it twice
+                else:
+                    demisto.debug(f"Skipping this step, max number of pull alerts already reached"
+                                  f"({save_alert_count} > {max_fetch_alerts}) for the incident {inc.get('id')} !")
+        inc_last_update = arg_to_datetime(inc["lastUpdated"])
+        if inc_last_update and last_update_format:
+            demisto.debug(f"Incident {inc.get('id')} - "
+                          f"Last run {last_update_format.timestamp()} - Last updated {inc_last_update.timestamp()} - "
+                          f"Need update => {last_update_format.timestamp() < inc_last_update.timestamp()}")
+            if last_update_format.timestamp() < inc_last_update.timestamp():
+                modified_incidents_ids.append(inc.get("id"))
+                continue  # if added no need to do it twice
```

Comment:
Yes because in the first condition it only check if the incident alertCount and eventCount changed, whereas in the second part it checked if the incident changed using lastUpdated field.
So we don't need to change that